### ISSUE TITLE
[usability] sync debugging

### ISSUE THREAD

**Issue by kwladyka** [argoproj/argo-cd#1566]
I modified the issue into usability issue. It shows better the point when can read context.

The issue:
kustomization.yaml didn't add Application resource, but added ConfigMap. When I moved file into the same directory it started to work so I assumed it can be something with `../base` directory which point to directory on higher level. I was feeling something is wrong with this assumption, but there is no way for debugging.  I solved it only, because Alexander told me argo cd use by default older kustomize version.

Solution?
+ In my personal feeling I would like to see all output to be sure what is happening under the hood. I prefer this way vs see kustomize version. Another time I will have another issue, we can add additional fields forever. See output for sync solve in for now and for the future.
+ add info about it in documentation

---
OLD DESCRIPTION

**Describe the bug**
```
tree
.
├── argocd
│   ├── auth
│   │   ├── base
│   │   │   ├── auth-provider
│   │   │   │   ├── deployments.yaml
│   │   │   │   ├── ingress.yaml
│   │   │   │   └── services.yaml
│   │   │   ├── kustomization.yaml
│   │   │   └── ory-hydra
│   │   │       ├── deployments.yaml
│   │   │       ├── ingress.yaml
│   │   │       └── services.yaml
│   │   ├── production
│   │   │   └── kustomization.yaml
│   │   └── staging
│   │       ├── add-image-pull-policy-always.yaml
│   │       ├── image-pull-policy.yaml
│   │       └── kustomization.yaml
│   └── cluster
│       ├── base
│       │   ├── auth.yaml
│       │   ├── kustomization.yaml
│       │   └── transformer-config.yaml
│       ├── production
│       │   └── kustomization.yaml
│       └── staging
│           └── kustomization.yaml
└── readme.md
```

argocd/cluster/staging
```
configMapGenerator:
- name: global
  literals:
  - ENVIRONMENT=staging

bases:
- ../base
```

Bugs:
1) Argocd doesn't add resources from `../base` at all. I guess it can't open `../` folders. (I can be totally wrong about a reason)
2) Argocd show 0 info about errors in such case. Everything is green. Somebody can miss it.
3) There is no way to debug it. I had to guess by trying.

**To Reproduce**
Use `../base` in `kustomize.yaml` to overlay another `kustomize.yaml`.

**Expected behavior**
Kustomize work as on local machine. So it loads overlays from path on higher levels `../` as it is idea of overlays.

**Comment by kwladyka** [comment]
I found https://argoproj.github.io/argo-cd/operator-manual/application.yaml

Please don't go into this direction about `kustomize` settings. Kustomize use overlays for it. Do not implement another type of overlay for argocd for kustomize. Let us use native kustomize solutions.

**Comment by twz123** [comment]
I think those overrides are just exposing the functionality of `kustomize edit`, which is nothing Argo CD specific, and still "kustomize native", if you will.

There are certainly use cases that justify the ability to specify some overrides. One of them which comes to my mind is a CICD pipeline that deploys the builds for each open PR in parallel. It would be a bit cumbersome to create an extra one-shot/throw-away git commit for each ephemeral build/deploy cycle triggered by the CI.

**Comment by kwladyka** [comment]
Not sure what you mean. Anyway it became a little off topic. The issue is about bugs. Probably my fault I mentioned about this in this context. Just I hope I will be able to use kustomize.yaml overlays for settings such things instead add this parameters in argocd reosurce. Anyway I can't use `bases` so it doesn't help to solve the issue.

**Comment by kwladyka** [comment]
Solved!

Argo CD use older version of kustomize by default.

Add this to `kustomize.yaml`:

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
```

**Comment by alexec** [comment]
The workaround for this is to make sure your Kustomize manifests have the `apiVersion/kind` fields. Argo CD uses these to detect if they're Kustomize 2. I think we should at least add some docs on this. Maybe default to k2 for now on.
